### PR TITLE
Selfmain changes

### PR DESCRIPTION
**Pull request by keith-packard**
Compiling the loader for SAMD21 generated an error as the code was reading past the end of the 'fuses' array in src/selfmain.c:122

These changes make the compiler happy, but I have not verified that they will actually do the right thing.

### CODE REVIEW

**Comment by dhalbert** [comment]
Thanks for finding this. I'll review it in detail. We are using gcc 7.3.1 and it didn't catch this problem. Which compiler did you use?

**Comment by keith-packard** [comment]
I'm using the Debian packaged embedded arm compiler, which is 8.3.1 (20190703). That takes a few more tricks to use as I don't package newlib-nano for debian, only picolibc.

File: src/selfmain.c [hunk 1]
```diff
@@ -117,9 +117,14 @@ void setBootProt(int v) {
...
+#define UPDATE_COUNT 2
+#elif defined(SAMD51)
+#define UPDATE_COUNT 16
+#endif
+    for (int i = 0; i < endWriteIndex; i += UPDATE_COUNT) {
+        uint32_t *const qwBlockAddr = ((uint32_t *const)NVM_FUSE_ADDR) + i;
+        memcpy(qwBlockAddr, &fuses[i], UPDATE_COUNT);
         #if defined(SAMD21)
             exec_cmdaddr(NVMCTRL_CTRLA_CMD_WAP, qwBlockAddr);
         #elif defined(SAMD51)
```
Comment (dhalbert) [inline src/selfmain.c]: This looks good, but the `qwBlockAddr` variable name is misleading for the SAMD21 case. Could you rewrite this as a single `#if`-`#elif`, and then just copy the loop into each arm of the `#if`, with the proper `i += ...` and the proper `exec_cmdaddr()`? I think that will be easier to read. Thanks.

Comment (keith-packard) [reply src/selfmain.c]: How about this -- I split the SAMD21 code into a non-looping bit as there are always 8 bytes to write in that case.

Comment (dhalbert) [reply src/selfmain.c]: Looks good, thank you!

**Pull request merged**